src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -615,15 +615,27 @@ public function isAbsolutePath($file)
      * @param string $dir    The directory where the temporary filename will be created
      * @param string $prefix The prefix of the generated temporary filename
      *                       Note: Windows uses only the first three characters of prefix
+     * @param string $suffix The suffix of the generated temporary filename
      *
      * @return string The new temporary filename (with path), or throw an exception on failure
      */
-    public function tempnam($dir, $prefix)
+    public function tempnam($dir, $prefix/*, string $suffix = ''*/)
     {
+        if (\func_num_args() < 3 && __CLASS__ !== \get_class($this) && __CLASS__ !== (new \ReflectionMethod($this, __FUNCTION__))->getDeclaringClass()->getName() && !$this instanceof \PHPUnit\Framework\MockObject\MockObject && !$this instanceof \Prophecy\Prophecy\ProphecySubjectInterface) {
```

Comment:
you are testing that the method is *called* with 3 arguments, not that it is *defined* with 3 arguments. Calling the parent method without passing an explicit suffix is totally fine.

So your deprecation message is wrong (it does not complain about what it tests) and not needed.
And a deprecation about the definition of the argument is not needed here. DebugClassLoader already does it.

Reply:
@stof Are you saying I can remove this block entirely with no other change? That having an extra argument is not a BC break at all? Why does this say otherwise? https://symfony.com/doc/current/contributing/code/bc.html#changing-classes

Public methods: Add argument with a default value
Allowed: No